pkg/cloud_provider/file/file.go

```diff
@@ -251,6 +255,64 @@ func (manager *gcfsServiceManager) ListInstances(ctx context.Context, obj *Servi
 	return activeInstances, nil
 }
 
+func (manager *gcfsServiceManager) ResizeInstance(ctx context.Context, obj *ServiceInstance) (*ServiceInstance, error) {
+	instanceuri := instanceURI(obj.Project, obj.Location, obj.Name)
+	instance, err := manager.GetInstance(ctx, obj)
+	if err != nil {
+		glog.Errorf("Failed to get instance %s for resize operation, error %v", instanceuri, err)
+		return nil, err
+	}
+
+	// High Scale tier supports shrink of capacity. However CSI spec does not support it.
+	if obj.Volume.SizeBytes <= instance.Volume.SizeBytes {
+		return instance, nil
+	}
+
```

Comment:
So regular tier suports changes in 1G increments, and high scale in 10T increments:
https://cloud.google.com/filestore/docs/editing-instances

I don't know if the API errors or rounds if you try to change in a smaller increment. Any ideas on how to handle that?

Reply:
Thats a good question. The current spec for Fileshare only provides CapacityGb option (https://cloud.google.com/filestore/docs/reference/rest/v1beta1/projects.locations.instances#fileshareconfig). i.e we have option to either do a floor or ceil of the requested bytes. Current implementation does a ceil.
I verified that, say we increase a PVC size to from 1Tb to 1Tb and another 500 mb, the driver sends a request for 1Tb + 1Gb, and the filestore edit API succeeds. This means now PVC reports 1101109657600 bytes (~1025.5 gb) while PV reports 1026 Gb.

If we go with floor option, that seems more consistent. the 1Tb + 500mb request would be floor to 1Tb and no resize would be done. I see PD does a floor too.(https://sourcegraph.com/github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver@82298d110dfe8acddce893825b553a62c7fb37b9/-/blob/pkg/gce-cloud-provider/compute/gce-compute.go#L663)

Regarding high scale tier, I dont have a setup to test. I expect any edits <10Tb should return an error. So no extra logic should be needed in the driver, and let filestore give back the appropriate error. But will confirm with the high scale team.

Reply:
Ok, makes sense. Generally reporting back filestore error makes sense, we shouldn't encode their logic in our driver.